lib/_http_agent.js

```diff
@@ -152,7 +152,12 @@ Agent.prototype.addRequest = function addRequest(req, options, port/*legacy*/,
     options.servername = options.host;
     const hostHeader = req.getHeader('host');
     if (hostHeader) {
-      options.servername = hostHeader.replace(/:.*$/, '');
+      // abc => abc
+      // abc:123 => abc
+      // [::1] => ::1
+      // [::1]:123 => ::1
+      options.servername = hostHeader.replace(/:[^\]]+$/, '')
```

Comment:
Why not combine both regexps for efficiency?:

```js
options.servername = hostHeader.replace(/^(?:\[([^\]]*)\].*)|(?:([^:]*).*)$/, '$1$2');
```

I tried a few different variations on the above regexps, and that one seemed to execute the fastest, at least on node v8.3.0.